### PR TITLE
codec: Move length delimited codec from tokio to tokio-codec

### PR DESCRIPTION
**Pull request by 95th**

### CODE REVIEW

File: tokio-codec/src/length_delimited.rs [hunk 1]
```diff
@@ -15,8 +15,8 @@
 //! byte stream into a stream of frames.
 //!
 //! ```
-//! use tokio::io::{AsyncRead, AsyncWrite};
-//! use tokio::codec::*;
+//! use tokio_io::{AsyncRead, AsyncWrite};
+//! use tokio_codec::*;
```
Comment (LucioFranco) [inline tokio-codec/src/length_delimited.rs]: minor nit: even though this was copied can we update this to not use a glob import?

File: tokio-codec/src/length_delimited.rs [hunk 2]
```diff
@@ -41,9 +41,9 @@
 //! ```
 //! #![feature(async_await)]
 //!
-//! use tokio::io::{AsyncRead, AsyncWrite};
-//! use tokio::codec::*;
-//! use tokio::prelude::*;
+//! use tokio_io::{AsyncRead, AsyncWrite};
+//! use tokio_codec::*;
```
Comment (LucioFranco) [inline tokio-codec/src/length_delimited.rs]: same here

File: tokio-codec/src/length_delimited.rs [hunk 3]
```diff
@@ -603,8 +607,8 @@ impl Builder {
     /// # Examples
     ///
     /// ```
-    /// # use tokio::io::AsyncRead;
-    /// use tokio::codec::length_delimited::Builder;
+    /// # use tokio_io::AsyncRead;
+    /// use tokio_codec::length_delimited::Builder;
```
Comment (LucioFranco) [inline tokio-codec/src/length_delimited.rs]: these examples should probably use `LenthDelimited::builder()`

File: tokio-codec/src/lib.rs [hunk 4]
```diff
@@ -31,4 +32,5 @@ pub use crate::encoder::Encoder;
 pub use crate::framed::{Framed, FramedParts};
 pub use crate::framed_read::FramedRead;
 pub use crate::framed_write::FramedWrite;
+pub use crate::length_delimited::{FrameTooBig, LengthDelimitedCodec};
```
Comment (LucioFranco) [inline tokio-codec/src/lib.rs]: I wonder if we should rename this `FrameTooBig` to a better error name similar to the `LinesCodecError`? What do you think?

Comment (95th) [reply tokio-codec/src/lib.rs]: I agree but something like `LengthDelimitedCodecError` is too long IMO.

Comment (LucioFranco) [reply tokio-codec/src/lib.rs]: Meh, I think for how often people might use it, I think its fine.

Comment (95th) [reply tokio-codec/src/lib.rs]: Ok, renaming to `LengthDelimitedCodecError` then

File: tokio-codec/src/length_delimited.rs (hunk 1 above)
Comment (95th) [reply tokio-codec/src/length_delimited.rs]: Done

File: tokio-codec/src/length_delimited.rs (hunk 2 above)
Comment (95th) [reply tokio-codec/src/length_delimited.rs]: Done

File: tokio-codec/src/length_delimited.rs (hunk 3 above)
Comment (95th) [reply tokio-codec/src/length_delimited.rs]: Done

File: tokio-codec/src/lib.rs (hunk 4 above)
Comment (95th) [reply tokio-codec/src/lib.rs]: Done

**Comment by LucioFranco** [comment]
@95th one last thing, looks like well need to bring over this test https://github.com/tokio-rs/tokio/blob/master/tokio/tests/length_delimited.rs

**Comment by 95th** [comment]
Doh. you're right. I'll move it

**Comment by 95th** [comment]
Done

**Comment by LucioFranco** [comment]
one last thing is to update the tests, we added a broken feature flag to the test so that we didnt have to worry about it when compiling. https://github.com/tokio-rs/tokio/blob/0ff295222f5dc8b23390f43966d482f75f63d9d5/tokio-codec/tests/length_delimited.rs#L1 that will need to be removed and the tests will need to be updated.

**Comment by LucioFranco** [comment]
once that is done, then I think this is ready to be shipped. Thanks!

**Comment by carllerche** [comment]
@95th @LucioFranco I have an in-flight PR that has already updated the tests. It would be easiest if this PR didn't touch the test file. Then, when I merge master, I can move the test file.

**Comment by 95th** [comment]
@carllerche Done

**Pull request merged**